coronavirus_biogrid.py

```diff
@@ -17,35 +17,67 @@
 import metadata
 from datetime import date
 import zipfile
+from atomwrappers import *
+from biogrid import checkdisc
 
-def checkdisc(diction, key, value):
-  try:
-    diction.setdefault(key,[]).append(value)
-  except KeyError:
-    return "key error"
 
-def evaLink(node1, node1_type, node2, node2_type, predicate, prefix1="", prefix2="",symmetric=False, stv=""):
+def evaLink(node1, node1_type, node2, node2_type, predicate, prefix1="", prefix2="", symmetric=False, stv=""):
     if not (str(node1) in ["-", "nan"] or str(node2) in ["-", "nan"]):
         if symmetric:
             list_type = "SetLink"
         else:
             list_type = "ListLink"
         return ("(EvaluationLink {}\n".format(stv) +
-            "\t (PredicateNode \""+ predicate + "\")\n" +
-            "\t ({} \n".format(list_type) +
-            "\t\t ({}".format(node1_type) + " \"" + prefix1 + str(node1) + "\")\n" +
-            "\t\t ({}".format(node2_type) + " \"" + prefix2 + str(node2) + "\")))\n" )
+                "\t (PredicateNode \"" + predicate + "\")\n" +
+                "\t ({} \n".format(list_type) +
+                "\t\t ({}".format(node1_type) + " \"" + prefix1 + str(node1) + "\")\n" +
+                "\t\t ({}".format(node2_type) + " \"" + prefix2 + str(node2) + "\")))\n")
     else:
         return ""
 
+
 def member(node1, node1_type, node2, node2_type, prefix1="", prefix2=""):
     if not (str(node1) in ["-", "nan"] or str(node2) in ["-", "nan"]):
-        return ('(MemberLink\n' + 
-                '\t({} "'.format(node1_type) + prefix1 + str(node1) + '")\n'+
+        return ('(MemberLink\n' +
+                '\t({} "'.format(node1_type) + prefix1 + str(node1) + '")\n' +
                 '\t({} "'.format(node2_type) + prefix2 + str(node2) + '"))\n')
     else:
         return ""
 
+
+def add_taxonomy(taxonomy_id, node, fp):
+    member_ln = CMemberLink(node, CConceptNode("ncbi:" + str(taxonomy_id)))
+    fp.write(member_ln.recursive_print() + "\n")
+
+
+def add_protein_interaction(proteins_lst, prot_node_1, gene_node_1, prot_node_2, gene_node_2, bio_id_1, bio_id_2, fp):
+    if not prot_node_1.name in proteins_lst:
+        express_ln = CEvaluationLink(CPredicateNode("expresses"), CListLink(gene_node_1, prot_node_1))
```

Comment:
I suggest to make a function from this code so that the code would look like that:

```
 if not prot_node_1.name in proteins_lst:
          proteins_lst += define_protein(fp, gene_node_1, prot_node_1)
```